---
title: "Course Introduction"
author: "Mark Dunning"
date: "28/03/2016"
output: slidy_presentation

---

# Welcome!


## About us

![us](../images/instructors.png)

- Mark
    + Cancer Research Uk, Cambridge
    + @DrMarkDunning
    + `mark.dunning@cruk.cam.ac.uk`
    
- Tom
    + Medical Research Council, Clinical Sciences Centre, London
    + `thomas.carroll@csc.mrc.ac.uk`
- Nuno
    + Instituto de Medicina Molecular, Lisboa
    + `nmorais@medicina.ulisboa.pt`


## Admin


## About the Course

- We will tell about '*best practice*' tools that we use in daily work as Bioinformaticians
- You will (probably) not come away being an expert
- We cannot teach you everything about NGS data
    + plus, it is a fast-moving field
- RNA and ChIP only
    + much of the initial processing is the same for other assays
- However, we hope that you will
    + Understand how your data are processed
    + Be able to explore your data - no programming required
    + Increase confidence with R and Bioconductor
    + Be able to explore new technologies, methods, tools as they come out

## Further disclaimer

![fisher](../images/fisher.jpg)

> To consult the statistician after an experiment is finished is often merely to ask him to conduct a post mortem examination. He can perhaps say what the experiment died of.‚Äù. R.A. Fisher, 1938

If you haven't designed your experiment properly, then all the Bioinformatics we teach you won't help: Consult with your local statistician - preferably **not the day before your grant is due!!!!**

<iframe width="420" height="315" src="https://www.youtube.com/embed/Hz1fyhVOjr4" frameborder="0" allowfullscreen></iframe>

- We have some [materials](http://bioinformatics-core-shared-training.github.io/experimental-design/) you can look at


# Course Outline

## Day 1

- Recap of R
- Introduce the Bioconductor project
- Hands-on experience with NGS data
    + IGV
    + FastQC

## Day 2

- Data structures for NGS analysis in R
- Statistical theory behind RNA-seq analysis
- RNA-seq intro
- Aligning and counting for RNA-seq

## Day 3

- Differential expression analysis for RNA-seq
- Annotating RNA-seq results
- Downstream analysis of RNA-seq

## Day 4

- Downstream analysis of RNA-seq
- Intro to ChIP-seq
- QA and analysis of ChIP data
- Wrap-up

# Historical context

## Cast your minds back a few years..

![array-summary](../images/array-summary.png)

## Plenty of success stories with microarrays

![array-achievements](../images/array-achievements.png)

## What did we learn from arrays?

- Experimental Design; despite this fancy new technolgy, if we don't design the experiments properly we won't get meaningful conclusions
- Quality assessment; Yes, NGS experiments can still go wrong!
- Normalisation; NGS data come with their own set of biases and error that need to be accounted for
- Stats; testing for RNA-seq is built-upon the knowledge from microarrays
- Plenty of tools and workflows were established.
- ***Don't forget about arrays; the data are all out there somewhere waiting to be discovered and explored***

## Reproducibility is key

![duke-scandal](../images/rep-research-nyt.png)

Two Biostatiscians (later termed '*Forensic Bioinformaticians*') from M.D. Anderson used R extensively during their re-analysis and investigation of a Clinical Prognostication paper from Duke. The subsequent [scandal](https://www.youtube.com/watch?v=W5sZTNPMQRM) put Reproducible Research on the map.

Keith Baggerly's talk from Cambridge in 2010 is highy-recommended.

<iframe width="420" height="315" src="https://www.youtube.com/embed/7gYIs7uYbMo" frameborder="0" allowfullscreen></iframe>

## Advantages of R

![NYT](../images/NYTimes_R_Article.png)

The R programming language is now recognised beyond the academic community as an effect solution for data analysis and visualisation. [Notable users of R](http://www.revolutionanalytics.com/companies-using-r) include [Facebook](http://blog.revolutionanalytics.com/2010/12/analysis-of-facebook-status-updates.html), [google](http://blog.revolutionanalytics.com/2009/05/google-using-r-to-analyze-effectiveness-of-tv-ads.html), [Microsoft](http://blog.revolutionanalytics.com/2014/05/microsoft-uses-r-for-xbox-matchmaking.html) (who recently [invested](http://blogs.microsoft.com/blog/2015/01/23/microsoft-acquire-revolution-analytics-help-customers-find-big-data-value-advanced-statistical-analysis/) in a commerical provider of R), and the [New York Times](http://blog.revolutionanalytics.com/2011/03/how-the-new-york-times-uses-r-for-data-visualization.html).

## Key features

- Open-source
- Cross-platform
- Access to existing visualisation / statistical tools
- Flexibility
- Visualisation and interactivity
- Add-ons for many fields of research
- Facilitating ***Reproducible Research***


# Crash-course in R



## Support for R

- Online forums such as [Stack Overflow](http://stackoverflow.com/questions/tagged/r) regularly feature R
- [Blogs](http://www.r-bloggers.com/)
- Local [user groups](http://blog.revolutionanalytics.com/local-r-groups.html) 
- Documentation via `?` or `help.start()`


## RStudio

![RStudio](http://www.rstudio.com/wp-content/uploads/2014/03/blue-125.png)

- Rstudio is a free environment for R
- Convenient menus to access scripts, display plots
- Still need to use *command-line* to get things done
- Developed by some of the leading R programmers

## R recap

R can do simple numerical calculations
```{r}
2  + 2
sqrt(25)
```

Here, `sqrt` is a *function* and the number 25 was used as an *argument* to the function. Functions can have multiple arguments

## Variables

We can save the result of a computation as a *variable* using the *assignment* operator `<-`

```{r}
x <- sqrt(25)
x + 5
y <- x +5
y
```

## Vectors

A vector can be used to combine multiple values. The resulting object is indexed and particular values can be queried using the `[]` operator

```{r}
vec <- c(1,2,3,6)
vec[1]
```

## Vectors

Calculations can be performed on vectors

```{r}
vec*2
mean(vec)
sum(vec)
```

## Data frames

These can be used to represent familiar tabular (row and column) data

```{r}
df <- data.frame(A = c(1,2,3,6), B = c(7,8,10,12))
df
```

## Data frames

Don't need the same data *type* in each column

```{r}
df <- data.frame(A = c(1,2,3,6), 
                 B = month.name[c(7,8,10,12)])
df
```

##Data frames

We can subset data frames using the `[]`, but can specify row and column indices

```{r}
df[1,2]
df[2,1]
```

##Data frames

```{r}
df[1,]
df[,2]
```


Or leave the row or column index blank to get all rows and columns respectively

## Plotting

All your favourite types of plot can be created in R

```{r echo=FALSE}

par(mfrow=c(2,2))
barplot(VADeaths, beside = TRUE,
        col = c("lightblue", "mistyrose", "lightcyan",
                "lavender", "cornsilk"), ylim = c(0, 100))
boxplot(len ~ dose, data = ToothGrowth,
        boxwex = 0.25, at = 1:3 - 0.2,
        subset = supp == "VC", col = "yellow",
        main = "Guinea Pigs' Tooth Growth",
        xlab = "Vitamin C dose mg",
        ylab = "tooth length",
        xlim = c(0.5, 3.5), ylim = c(0, 35), yaxs = "i")
boxplot(len ~ dose, data = ToothGrowth, add = TRUE,boxwex = 0.25, at = 1:3 + 0.2,subset = supp == "OJ", col = "orange")
legend(2, 9, c("Ascorbic acid", "Orange juice"),fill = c("yellow", "orange"))
set.seed(14)
x <- rchisq(100, df = 4)
## if you really insist on using hist() ... :
hist(x, freq = FALSE, ylim = c(0, 0.2))
curve(dchisq(x, df = 4), col = 2, lty = 2, lwd = 2, add = TRUE)
pie(c(Sky = 78, "Sunny side of pyramid" = 17, "Shady side of pyramid" = 5),
    init.angle = 315, col = c("deepskyblue", "yellow", "yellow3"), border = FALSE)
```

## Plotting

- Simple plots are supported in the *base* distribution of R (what you get automatically when you download R). 
    + `boxplot`, `hist`, `barplot`,... all of which are extensions of the basic `plot` function
- Many different customisations are possible
    + colour, overlay points / text, legends, multi-panel figures
- We will show how some of these plots can be used to inform us about the quality of NGS data, and to visualise our results.
- References..
    + [Introductory R course](http://cambiotraining.github.io/r-intro/)
    + [Quick-R](http://www.statmethods.net/graphs/index.html)

## The Bioconductor project

![BioC](../images/logo_bioconductor.png)

-  Packages analyse all kinds of Genomic data (>800)
- Compulsory documentation (*vignettes*) for each package
- 6-month release cycle
- [Course Materials](http://bioconductor.org/help/course-materials/)
- [Example data](http://bioconductor.org/packages/release/BiocViews.html#___ExperimentData) and [workflows](http://bioconductor.org/help/workflows/)
- Common, re-usable framework and functionality
- [Available Support](https://support.bioconductor.org/)
    + Often you will be able to interact with the package maintainers / developers and other power-users of the project software
- Annual conferences in U.S and Europe
    - The last European conference was in [Cambridge](https://sites.google.com/site/eurobioc2015/)
    
## The Bioconductor project    
    
    
Many of the packages are by well-respected authors and get lots of citations.

![citations](../images/citations.png)

##Downloading a package

Each package has its own landing page. e.g. http://bioconductor.org/packages/release/bioc/html/beadarray.html. Here you'll find;

- Installation script (will install all dependancies)
- Vignettes and manuals
- Details of package maintainer
- After downloading, you can load using the `library` function. e.g. `library(beadarray)`
- Only need to download once for each version of R
- CRAN packages installed by `install.packages`
- What packages to install?
    + [METACRAN](http://www.r-pkg.org/) can help




## Introducing the practical

- Refresh your memory of R skills
    + reading data
    + subsetting data
    + plotting
