# NDARC16
4-day course at Instituto Gulbenkian de Ciência

# Overview

High-throughput technologies such as next generation sequencing (NGS) can routinely produce massive amounts of data. These technologies allow us to describe all variants in a genome or to detect the whole set of transcripts that are present in a cell or tissue. However, such datasets pose new challenges in the way the data have to be analyzed, annotated and interpreted which are not trivial and are daunting to the wet-lab biologist. This course covers state-of-the-art and best-practice tools for NGS RNA-seq and ChIP-seq data analysis, which are of major relevance in today’s genomic and gene expression studies. 

# Methods

The course is comprised of practical exercises preceded by short lectures. Exercises will be conducted primarily in the R programming language.

# Target Audiences

Enthusiastic and motivated wet-lab biologists who want to gain more of an understanding of NGS data and eventually progress to analysing their own data

# Pre-requisites
There is a lot of material to cover in the course, so we will assume that you are familiar with a few basics before you come. The tool that will we do most of the analysis in is R. There will be a short recap of the key concepts at the beginning of the course; however it will be beneficial if you are already familiar with how to read data into R, perform basic subset operations and produce simple plots. 

Several Online videos are available that cover this materials. For example

- http://shop.oreilly.com/product/0636920034834.do
- http://blog.revolutionanalytics.com/2012/12/coursera-videos.html
- http://bitesizebio.com/webinar/20600/beginners-introduction-to-r-statistical-software

Some introductory statistics will be also be assumed. See [Statistics at Square One](http://www.bmj.com/about-bmj/resources-readers/publications/statistics-square-one) for a good overview. 
